Fixed docker-compose packaging/building

sumdog opened the pull request:
Fixes #696 

This merge requests does a couple of things:

1. Removes the exposed MySQL port and instead, uses container linking variables
2. Handles configuration using docker env variables (setup in docker-runner.sh)
3. Waits to ensure mysql is up before running migrations

Liongold commented:
Please point this to the develop branch. Also, please re-check your commit since the Travis build is failing.

sumdog commented:
I've moved this to a develop branch pull request:

https://github.com/twostairs/paperwork/pull/699

I'm not sure why the build would be failing. The TravisCI logs show an npm/gulp issue, but nothing I changed in this patch should affect those components.
